controllers/external/util.go

```diff
@@ -31,17 +31,17 @@ import (
 )
 
 // Get uses the client and reference to get an external, unstructured object.
-func Get(ctx context.Context, c client.Reader, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+func Get(ctx context.Context, c client.Reader, ref *corev1.ObjectReference) (*unstructured.Unstructured, error) {
 	if ref == nil {
 		return nil, errors.Errorf("cannot get object - object reference not set")
 	}
 	obj := new(unstructured.Unstructured)
 	obj.SetAPIVersion(ref.APIVersion)
 	obj.SetKind(ref.Kind)
 	obj.SetName(ref.Name)
-	key := client.ObjectKey{Name: obj.GetName(), Namespace: namespace}
-	if err := c.Get(ctx, key, obj); err != nil {
-		return nil, errors.Wrapf(err, "failed to retrieve %s external object %q/%q", obj.GetKind(), key.Namespace, key.Name)
+	obj.SetNamespace(ref.Namespace)
+	if err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve %s external object %q/%q", obj.GetKind(), ref.Namespace, ref.Name)
```

Comment:
```suggestion
		return nil, errors.Wrapf(err, "failed to retrieve %s %s/%s", obj.GetKind(), ref.Namespace, ref.Name)
```
While we're here, let's align this to how we produce these errors nowadays :)

Reply:
branching from this PR, but it'd be nice to align on client.ObjectKeyFromObject(resource).String()

Reply:
I think if we align we should align to klog.KRef. We use it everywhere in logs